src/imap.rs

```diff
@@ -1910,6 +1926,15 @@ pub(crate) async fn prefetch_should_download(
     // We do not know the Message-ID or the Message-ID is missing (in this case, we create one in
     // the further process).
 
+    // Never download messages directly from the spam folder.
+    // If the sender is known, the message will be moved to the Inbox or Mvbox
+    // and then we download the message from there.
+    // Also see `spam_target_folder()`.
+    if context.is_spam_folder(folder).await? {
```

Comment:
Instead of passing the folder into `prefetch_should_download`, I think this check should be done before calling `prefetch_should_download` here:
https://github.com/deltachat/deltachat-core-rust/blob/cccdc51ad40f7c28447b00aff685f108a29976aa/src/imap.rs#L754

Not a big deal if Bcc-selfs are not marked as seen if they get into Spam folder.